### PR TITLE
[fr] VIRGULE_DEBUT_DE_PHRASE_3

### PR DESCRIPTION
**Pull request by remymartinDev**
this is a first shot
I'm sure that the 3 rules are factorisable so I'll rework the rulegroup to optimize the code

### CODE REVIEW

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml [hunk 1]
```diff
@@ -39150,6 +39150,100 @@ though, and has since been slightly modified)-->
...
+                    <marker>
+                        <token regexp="yes">nuits?|journées?|vacances|après-midi|matinées?
+                            <exception postag="P.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="N.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="D.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="Z.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="C coor.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="R pers obj.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="J.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="A.*" postag_regexp="yes" scope="next"/>
```
Comment (udomai) [inline languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:39197]: You can tidy this up:

`<exception postag="[PNDZJA].*|C coor.*|R pers obj.*" postag_regexp="yes" scope="next"/>`.

Or you can go right ahead and say what you want the next token to be (`R pers suj`?). Do you really want to exclude all the others?

"Bonjour mon ami !" seems to be missing a comma, as does "Bonne nuit camarade !"

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml [hunk 2]
```diff
@@ -39150,6 +39150,100 @@ though, and has since been slightly modified)-->
...
+                <example correction="après-midi,">Bonne <marker>après-midi</marker> je rentre chez moi.</example>
+                <example>Bonnes vacances!</example>
+                <example>Bonne nuit chez vous. (P)</example>
+                <example>Bonne nuit camarade. (N)</example>
+                <example>Bonne nuit les amis. (D)</example>
+                <example>Bonne nuit Jean. (Z)</example>
+                <example>Bonne nuit et bienvenue. (C coor)</example>
+                <example>Bonne nuit toi. (R pers obj)</example>
+                <example>Bonne nuit chers amis. (J)</example>
+                <example>Bonne nuit toutes et tous. (A)</example>
```
Comment (udomai) [inline languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:39213]: This kind of technical remark cannot be included in an example (examples will be visible to users at some point, so we cannot put any technical stuff in there). If you want to include a comment, you can do that in a `<!-- … -->` after the example.

**Review by udomai — CHANGES_REQUESTED** [review]

**Comment by remymartinDev** [comment]
I have to work more on this, because of N Z D, to find a way to make a difference between : 
- Bonjour, Jean.
- Bonjour Jean, je répond à ton email [...]

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml [hunk 3]
```diff
@@ -39150,6 +39150,90 @@ though, and has since been slightly modified)-->
                 <example correction="vraisemblance,">Selon toute <marker>vraisemblance</marker> cet évènement n'était pas le fruit du hasard.</example>
             </rule>
         </rulegroup>
+        <rulegroup id="VIRGULE_DEBUT_DE_PHRASE_3" name="virgule debut de phrase" default="temp_off">
+            <rule>
+                <antipattern>
+                    <token postag="SENT_START"/>
+                    <token regexp="yes">bon(jour|soir)|salut|enchantée?s?|bienvenue?s?</token>
+                    <token postag="[PNDZJA].*|C coor.*|R pers obj.*" postag_regexp="yes"/>
+                    <token postag=".*" postag_regexp="yes" min="0" max ="6"/>
```
Comment (udomai) [inline languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:39159]: `max ="6"` → `max="6"`

**Review by udomai — CHANGES_REQUESTED** [review]

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml (hunk 3 above)
Comment (udomai) [reply languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:39159]: `postag=".*" postag_regexp="yes"` → tu peux supprimer cela

**Review by udomai — CHANGES_REQUESTED** [review]

File: languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml [hunk 4]
```diff
@@ -39150,6 +39150,90 @@ though, and has since been slightly modified)-->
...
+                <example>Bonjour chez vous.</example>
+                <example>Bonjour et bienvenue.</example>
+            </rule>
+            <rule>
+                <antipattern>
+                    <token postag="SENT_START"/>
+                    <token regexp="yes">bonnes?</token>
+                    <token regexp="yes">nuits?|journées?|vacances|après-midi|matinées?</token>
+                    <token postag="[PNDZJA].*|C coor.*|R pers obj.*" postag_regexp="yes"/>
+                    <token min="0" max ="6"/>
```
Comment (udomai) [inline languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml:39186]: `max ="6"` → `max="6"`

**Review by udomai — CHANGES_REQUESTED** [review]

**Pull request merged**